core/src/main/java/com/microsoft/applicationinsights/web/internal/correlation/CdsProfileFetcher.java

```diff
@@ -110,15 +112,12 @@ public ProfileFetcherResult fetchApplicationId(String instrumentationKey, Teleme
             Throwable cause = e.getCause();
             if (cause != null && cause instanceof SSLHandshakeException) {
                 URI uri =configuration.getEndpointProvider().getAppIdEndpointURL(instrumentationKey);
-                throw new FriendlyException("ApplicationInsights Java Agent failed to connect to CdsProfiler end point.",
-                        "Unable to find valid certification path to requested target.",
-                        "Please import the SSL certificate from "+ uri.getHost() +", into the java key store. "+
-                        "Learn more about importing the certificate here: https://go.microsoft.com/fwlink/?linkid=2151450",
-                        "This message is only logged the first time it occurs after startup.");
+                SSLUtil.throwSSLFriendlyException(uri.getHost());
```

Comment:
ah, sorry, with this, then don't need that strange `return null` below

```suggestion
                throw SSLUtil.newSSLFriendlyException(uri.getHost());
```